### ISSUE TITLE
File autocomplete freeze (large amount of files)

### ISSUE THREAD

**Issue by antigol** [jupyter/help#198]
Hi,

I have a directory with thousand files in it.
In jupyter-notebook, when I type `"/directory/` and "by mistake" press _tab_ jupyter freeze for 5 minutes.
Would it be possible to just don't autocomplete if the amount of files is above a certain threshold ?

**Comment by takluyver** [comment]
What platform are you on?

**Comment by antigol** [comment]
```
➜  ~ uname -a
Linux yoga 4.9.0-3-amd64 #1 SMP Debian 4.9.30-2+deb9u2 (2017-06-26) x86_64 GNU/Linux
➜  ~ jupyter-notebook --version
4.2.3
```

**Comment by takluyver** [comment]
And how many files? I've just made a folder with 1000 files, and it can still tab complete ~instantly.

**Comment by antigol** [comment]
60000

Le 14 août 2017 12:14 PM, "Thomas Kluyver" <notifications@github.com> a
écrit :

**Comment by takluyver** [comment]
Right, now I see it too. Yes, we should probably put a limit on the number of completions which are returned.

**Comment by takluyver** [comment]
See https://github.com/ipython/ipython/pull/10743

**Comment by antigol** [comment]
Thanks !
